### ISSUE TITLE
Failed examples

### ISSUE THREAD

**Issue by winfinit** [nodejs/nan#400]
I am trying to port examples one to one from node-addon-examples to 2.0.0, and ran into an issue with returning a function, instead of an object. Here is what i would like to happen:

```javascript
var addon = require('bindings')('addon');

addon(function(msg){
  console.log(msg); // 'hello world'
});
```

here is previous cpp code:
```cpp
#include <nan.h>

using namespace v8;

NAN_METHOD(RunCallback) {
  NanScope();

  Local<Function> cb = args[0].As<Function>();
  const unsigned argc = 1;
  Local<Value> argv[argc] = { NanNew("hello world") };
  NanMakeCallback(NanGetCurrentContext()->Global(), cb, argc, argv);

  NanReturnUndefined();
}

void Init(Handle<Object> exports, Handle<Object> module) {
  NODE_SET_METHOD(module, "exports", RunCallback);
}

NODE_MODULE(addon, Init)
```

now new NAN would look like this:
```cpp
#include <nan.h>

using namespace v8;

NAN_METHOD(RunCallback) {

  Local<Function> cb = info[0].As<Function>();
  const unsigned argc = 1;
  Local<Value> argv[argc] = { Nan::New<v8::String>("hello world").ToLocalChecked() };
  Nan::MakeCallback(info.This(), cb, argc, argv);

  info.GetReturnValue().SetUndefined();
}

NAN_MODULE_INIT(Init) {
  Nan::Set(target
    , Nan::New<v8::String>("exports").ToLocalChecked()
    , Nan::New<v8::FunctionTemplate>(RunCallback)->GetFunction()
  );
  //NAN_EXPORT(target, RunCallback);
}

NODE_MODULE(addon, Init)
```

however that would lead a return of an object with exposed function "exports" that would execute "RunCallback", which would make above javascript to change to:

```javascript
var addon = require('bindings')('addon');

console.log(addon);
addon.exports(function(msg){
  console.log(msg); // 'hello world'
});
```

and in case of NAN_EXPORT, function name would be equal to exposed function, i.e.: "RunCallback".

is there a way with NAN 2.0.0 to make first example working?

**Comment by kkoopa** [comment]
https://github.com/nodejs/nan/blob/master/doc/methods.md#api_nan_set_method

**Comment by winfinit** [comment]
from what i see, Nan::SetMethod requires three argument also, second being name of exported function, so this would lead to return in js same as Nan::Set

```cpp
NAN_MODULE_INIT(Init) {
  Nan::SetMethod(target
    , "exports"
    , RunCallback
  );
}
```

```javascript
var addon = require('bindings')('addon');
console.log(addon);
/* output:
{ exports: [Function: exports],
  path: '/Users/rjurkov/workspace/node-addon-examples/3_callbacks/nan_2.x/build/Release/addon.node' }
*/
```

with NAN 1.x same code outputs this:
```bash
{ [Function: exports]
  path: '/Users/rjurkov/workspace/node-addon-examples/3_callbacks/nan_1.x/build/Release/addon.node' }
```

**Comment by kkoopa** [comment]
Ah, you're lacking the module argument. It is not available in Node 0.8, so it is not included. Since you don't care about 0.8 support, you can hack up your own init function as `void Init(v8::Handle<v8::Object> exports, v8::Handle<v8::Value> module)` for all of v8 less than 4.5 and as taking `v8::Local` instead of `v8::Handle>` after that.

**Comment by winfinit** [comment]
just tried it before reading your reply! :) just posting here what you replied:

```javascript
void Init(Handle<Object> exports, Handle<Object> module) {
  Nan::SetMethod(module
    , "exports"
    , RunCallback
  );
}
```

thank you
